### ISSUE TITLE
Annotations should be allowed on same line when applied to receiver type of function

### ISSUE THREAD

**Issue by ned** [pinterest/ktlint#1812]
## Expected Rule behavior

This example

```kotlin
fun @Suppress("unused") MyReceiver.myFoo() {
    ...
}
```

shows the error `Annotations with parameters should all be placed on separate lines prior to the annotated construct (experimental:annotation)`.

Which can be fixed with

```kotlin
fun @Suppress("unused")
MyReceiver.myFoo() {
    ...
}
```

but I think that's worse than keeping everything on one line.

Note: this annotation applies to the receiver type, but not the function. So, in this example, IntelliJ will suppress warnings about the unused receiver type. But it will not suppress warnings about the function if it's unused.

## Additional information
* Current version of ktlint: 0.45.2 with `--experimental`

**Comment by matejdro** [comment]
Similarly, applying annotations to constructor should also be allowed:

```kotlin
class Test @Suppress("unused") constructor()
```

**Comment by paul-dingemans** [comment]
The annotation rule is indeed giving problems in situations like this. Also see https://github.com/pinterest/ktlint/issues/1813 in which similar problems were reported with a parameter type.

For now, I can not think of way, how annotations can be formatted in an automated way that is more or less acceptable to all.

Not only is this problematic with annotation with parameters but also in case the parameter type has multiple annotations, a similar problem arises:

fun demo(arg: @AnnotationA @Annotation @AnnotationC String) {}
It could be formatted as:

fun demo(
    arg:
        @AnnotationA
        @Annotation
        @AnnotationC
        String
) {}
Above looks more or less acceptable when having a lot of annotations or when having annotations with really long names. But with only two short annotation or one annotation with a parameter it looks ugly. Also, I do not expect this to happen a lot in normal code. So I think that the annotation rule should be restricted to the annotations which are preceding the function/class/declarations only and not be applied on inner annotations.

**Comment by paul-dingemans** [comment]
In ktlint `0.49.x` the `annotation` has been updated. From the viewpoint of consistency, it is chosen to format annotation in all cases in the same way.

Code below (based on examples mentioned before):
```
fun @Suppress("unused") MyReceiver.myFoo() {}
fun @Foo @Bar MyReceiver.myFoo() {}

class Test @Suppress("unused") constructor()
class Test @Foo @Bar constructor()

fun demo(arg: @Suppress("unused") String) {}
fun demo(arg: @Foo @Bar String) {}
```
is formatted since `0.49.x` (using code style `ktlint_official`) as follows:
```
fun
    @Suppress("unused")
    MyReceiver.myFoo() {}
fun
    @Foo @Bar
    MyReceiver.myFoo() {}

class Test
    @Suppress("unused")
    constructor()
class Test
    @Foo @Bar
    constructor()

fun demo(
    arg:
        @Suppress("unused")
        String,
) {}
fun demo(
    arg:
        @Foo @Bar
        String,
) {}
```

Annotations on the same line as the receiver type or the primary constructor will not be supported. The code produced by ktlint `0.49.x` will not be optimal in all cases. Formatting the annotations in this way is deterministic and consistent.
